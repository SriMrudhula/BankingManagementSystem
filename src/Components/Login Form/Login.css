.err{
    font-size: 10px;
    color: red;
}
.img2{
    background-image: url('../../Asserts/banking.jpg');
    height: 400px;
    background-repeat: no-repeat;
    background-size: cover;
    margin-top: -90px;
    width:100%;
    background-attachment: fixed;
}


.b3{
    display:inline-block;
    width: 350px;
    height: 400px;
    padding: 30px;
    background-color: rgba(0,0,0,0.5);
    background-size: 400px 480px;   
    margin-top:30px;
    position: relative;
     } 
     label{
       font-size: large;
        color: white;
        margin-left: 30px;
    }
    input{
        margin-left: 60px;
    }
    .heading{
        color: white;
        font-size: 30px;
    }
    .btn-primary{
        width:100px;
        height: 50px;
        border-radius: 15px;
    }
    