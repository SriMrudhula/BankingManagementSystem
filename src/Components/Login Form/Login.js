import React, { Component } from 'react';
import './Login.css';
import axios from 'axios';
import { Col, Row } from 'reactstrap';
import { Container } from '@material-ui/core';
class Login extends Component
{
    state={   
        username:"",
        password:"",
        x:"",
        users: []
    };
    componentDidMount(){
        axios.get("http://localhost:3000/users")
        .then(response =>{
            this.state.users=response.data
        })
        .catch(error=>{
            console.log(error);
        })
    }

    handleChange =event=>{
        this.setState({
            [event.target.name]:event.target.value
        });
    };
    
    handleSubmit= event =>{
        for(let i=0;i<this.state.users.length;i++)
        {
            console.log(this.state.users[i].username+" "+this.state.username);
            console.log(this.state.users[i].password+" "+this.state.password);
            if(this.state.users[i].username===this.state.username && this.state.users[i].password===this.state.password)
             {
                 this.state.x="successfully logged in";
                 localStorage.setItem("id",this.state.users[i].id);
                 localStorage.setItem("name",this.state.users[i].name);
                 localStorage.setItem("userName",this.state.users[i].username);
                 localStorage.setItem("password",this.state.users[i].password);
                 localStorage.setItem("address",this.state.users[i].address);  
                 localStorage.setItem("country",this.state.users[i].country);
                 localStorage.setItem("state",this.state.users[i].state);
                 localStorage.setItem("email",this.state.users[i].email);
                 localStorage.setItem("contact",this.state.users[i].contact); 
                 localStorage.setItem("dob",this.state.users[i].dob);
                 localStorage.setItem("accType",this.state.users[i].accType);
                 localStorage.setItem("branchName",this.state.users[i].branchName);
                 localStorage.setItem("initialDeposit",this.state.users[i].initialDeposit); 
                 localStorage.setItem("identificationProofType",this.state.users[i].identificationProofType); 
                 alert("Successfully Loggedin");
                 this.handleClick("home");  
                 break;
             }
             else
                 this.state.x="invalid user";             
             }
            if(this.state.x==="invalid user")
               alert("invalid");
        }
    handleRegister=()=>
    {
        this.handleClick("register");  
    }
    handleClick = event => {
        this.props.handleClick(event);     
    };   
    render()
    {
        return (
            <div className="img2">
            <div className="b3">
            <div className="heading">Login</div><br/><br/>
            <Container>
            <form onSubmit={this.handleSubmit} >
            <div className="form-group">
            <Row>
            <input name="username" type="text" placeholder="Enter UserName" className="form-control" value={this.state.username} onChange={this.handleChange} required/>
            </Row><br/>
            <Row>
            <input name="password" type="password" placeholder="Enter Password" className="form-control" value={this.state.password} onChange={this.handleChange} required/>
            
            </Row><br/>
            <button className="btn-primary">Login</button>
            </div>
            </form>
            </Container>
            Create an Account <button onClick={this.handleRegister}>Register</button> 
            </div>
            </div>
            )
        }
    }
export default Login;