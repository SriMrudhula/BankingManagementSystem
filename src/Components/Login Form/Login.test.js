import * as React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './Login';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
Enzyme.configure({ adapter: new Adapter() });
describe('Test when form is submitted', () => {
    it('should submit form data', () => {
        const testValues = {
            username: 'mno',
            password: 'mno',
        };
        let form = shallow(<Login />); 
        form.instance().handleSubmit = jest.fn();
        form.instance().handleSubmit();
        form.update(); 
        form.find('#loginsubmit').simulate('submit');


        expect(form.find('#loginsubmit').length).toEqual(1);
        expect(form.instance().handleSubmit).toHaveBeenCalled();
        // expect(form.instance().handleSubmit.toHaveBeCalledWith({username: testValues.username, password: testValues.password});
    });
    
    it('renders correctly', () => {
        const tree = renderer.create(<BrowserRouter><Login/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});
