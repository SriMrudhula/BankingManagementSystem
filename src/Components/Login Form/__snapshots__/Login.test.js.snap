// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test when form is submitted renders correctly 1`] = `
<div
  className="img2"
>
  <div
    className="b3"
  >
    <div
      className="heading"
    >
      Login
    </div>
    <br />
    <br />
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <div
            className="row"
          >
            <input
              className="form-control"
              name="username"
              onChange={[Function]}
              placeholder="Enter UserName"
              required={true}
              type="text"
              value=""
            />
          </div>
          <br />
          <div
            className="row"
          >
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="Enter Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <br />
          <button
            className="btn-primary"
            id="loginsubmit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
    Create an Account 
    <button
      onClick={[Function]}
    >
      Register
    </button>
  </div>
</div>
`;
